tests/unit/models/test_iterable_index_list.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+from pinecone import IterableIndexList
+from pinecone.core.client.models import IndexList
+
+@pytest.fixture
+def index_list_response():
+    return IndexList(
+            indexes=[
+                {
+                    "name": "test-index-1",
+                    "dimension": 2,
+                    "metric": "cosine",
+                    "capacity_mode": "serverless",
```

Comment:
Do these index models need to be updated to the new `spec` shape? Could we use `IndexModel` directly like below in `test_control.py`?

Reply:
Copilot let me down here 😂 

I'll update them. It doesn't really affect the way the tests work, but it makes the test better documentation if we use correct data.